cli/src/main/kotlin/commands/UploadResultToSw360Command.kt

```diff
@@ -0,0 +1,142 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.commands
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.core.requireObject
+import com.github.ajalt.clikt.parameters.options.convert
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.file
+
+import kotlin.time.measureTimedValue
+
+import org.eclipse.sw360.antenna.api.service.ServiceFactory
+import org.eclipse.sw360.antenna.sw360.client.adapter.SW360Connection
+import org.eclipse.sw360.antenna.sw360.client.adapter.SW360ConnectionFactory
+import org.eclipse.sw360.antenna.sw360.client.config.SW360ClientConfig
+import org.eclipse.sw360.antenna.sw360.client.rest.resource.releases.SW360Release
+import org.eclipse.sw360.antenna.sw360.client.utils.SW360ClientException
+
+import org.ossreviewtoolkit.GlobalOptions
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.config.Sw360StorageConfiguration
+import org.ossreviewtoolkit.model.readValue
+import org.ossreviewtoolkit.utils.expandTilde
+import org.ossreviewtoolkit.utils.formatSizeInMib
+import org.ossreviewtoolkit.utils.log
+import org.ossreviewtoolkit.utils.perf
+
+class UploadResultToSw360Command : CliktCommand(
+    name = "upload-result-to-sw360",
+    help = "Upload an ORT result to SW360.",
+    epilog = "EXPERIMENTAL: The command is still in development and usage will likely change in the near future. The " +
+            "command expects that a Sw360Storage for the scanner is configured."
+) {
+    private val ortFile by option(
+        "--ort-file", "-i",
+        help = "The ORT result file to read as input."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+        .inputGroup()
+
+    private val globalOptionsForSubcommands by requireObject<GlobalOptions>()
+
+    override fun run() {
+        val (ortResult, duration) = measureTimedValue { ortFile.readValue<OrtResult>() }
+
+        log.perf {
+            "Read ORT result from '${ortFile.name}' (${ortFile.formatSizeInMib}) in ${duration.inMilliseconds}ms."
+        }
+
+        val sw360Config = globalOptionsForSubcommands.config.scanner?.storages?.values
+            ?.filterIsInstance<Sw360StorageConfiguration>()?.singleOrNull()
+
+        requireNotNull(sw360Config) {
+            "No SW360 storage is configured for the scanner."
+        }
+
+        createSw360Connection(sw360Config)?.let { sw360Connection ->
```

Comment:
Instead of failing silently if the connection is `null`, and to reduce indention, I'd probably write this as (after doing the changes mentioned below):

    val sw360Config = SW360ClientConfig.createConfig(...)
    val sw360Connection = SW360ConnectionFactory().newConnection(sw360Config)

    requireNotNull(sw360Connection) {
        // ...
    }